{{cookiecutter.project_slug}}/CMakeLists.txt

```diff
@@ -22,8 +22,20 @@ set(BUILD_PYTHON ON CACHE BOOL "Enable building of Python bindings")
 set(BUILD_DOCS ON CACHE BOOL "Enable building of documentation")
 {%- endif %}
 
+{%- if cookiecutter.header_only == "No" %}
 # compile the library
 add_subdirectory(src)
+{% else %}
+# Add an interface target for our header-only library
+add_library({{ cookiecutter.project_slug }} INTERFACE)
+target_include_directories({{ cookiecutter.project_slug }} INTERFACE
+  $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/include/>
+  $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+)
+
+# Add an alias for use if this project is included as a subproject in another project
+add_library({{ cookiecutter.project_slug }}::{{ cookiecutter.project_slug }} ALIAS {{ cookiecutter.project_slug }})
```

Comment:
couldn't this also be done for the non header-only case?

Reply:
It is - it is just in the `src` subdirectory where the library is defined and which does not exist in the header-only case. This could be refactored to always happen in the top-level file.